src/process_utility.c

```diff
@@ -1108,13 +1107,19 @@ block_dropping_continuous_aggregates_without_cascade(ProcessUtilityArgs *args, D
 		name = get_rel_name(relid);
 
 		cagg = ts_continuous_agg_find_by_view_name(schema, name);
-		if (cagg == NULL)
-			continue;
-
-		if (ts_continuous_agg_view_type(&cagg->data, schema, name) == ContinuousAggUserView)
-			ereport(ERROR,
-					(errcode(ERRCODE_DEPENDENT_OBJECTS_STILL_EXIST),
-					 errmsg("dropping a continuous aggregate requires using CASCADE")));
+		if (cagg)
+		{
+			/* Add the materialization table to the arguments so that the
+			 * continuous aggregate and associated materialization table is
+			 * dropped together.
+			 *
+			 * If the table is missing, something is wrong, but we proceed
+			 * with dropping the view anyway since the user cannot get rid of
+			 * the broken view if we generate an error. */
+			Hypertable *ht = ts_hypertable_get_by_id(cagg->data.mat_hypertable_id);
```

Comment:
We are making an assumption here that the only dependent object on the cagg is the materialization hypertable. What happens in the case where we have a view defined on top of the cagg like: ` CREATE VIEW  joinview as select * from cagg, labels_tab where cagg.  = ...` . The CASCADE requirement can be dropped only if the dependent objects are internally created objects.

Reply:
Not sure I follow; this code looks up the materialized hypertable and adds it to the list of tables to be dropped. It make no assumptions on other objects and, in particular, does not allow dropping other (dependent) objects without using `CASCADE`.

Reply:
Thanks for the clarification. See that you also have a test case for it.